File: TrainingsNotebook/backend/src/main/java/com/trainings_notebook/backend/domain/Exercise.java
package com.trainings_notebook.backend.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity(name = "exercises")
@Table
public class Exercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    @NonNull
    private String name;

    @Column(name = "category")
    @Enumerated(EnumType.STRING)
    @NonNull
    private ExerciseCategories category;

    @Column(name = "description")
    private String description;

    @Column(name = "reps")
    private int reps;

    @Column(name = "duration")
    private int duration;

    @ManyToMany(mappedBy = "trainingExercises")
    private List<Training> trainings;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "workout_id")
    @JsonBackReference
    private Workout workout;


}

File: TrainingsNotebookAngular/src/app/models/Exercise.ts
import { FormGroup } from "@angular/forms";
import { Workout } from "./Workout";

export class Exercise {
   id: number;
   name: string;
   category: string;
   description: string;
   reps?: number;
   duration?: number;
   workout: Workout;
  

  static mapFormGroupObjectToExercise(formGroupObject: FormGroup) {
    let newExercise = new Exercise();
    newExercise.name = formGroupObject.get('name').value;
    newExercise.category = formGroupObject.get('category').value;
    newExercise.reps = formGroupObject.get('reps').value;
    newExercise.duration = formGroupObject.get('duration').value;
    newExercise.description = formGroupObject.get('description').value;
    console.log("worout: " + formGroupObject.get('workout').value);
    newExercise.workout = formGroupObject.get('workout').value;


    return newExercise;
  }

}